idb/postgres/internal/ledger_for_evaluator/ledger_for_evaluator_test.go

```diff
@@ -474,6 +474,99 @@ func TestLedgerForEvaluatorAccountAppTable(t *testing.T) {
 	assert.Equal(t, accountDataExpected, accountDataRet)
 }
 
+// Tests that queuying and reading from a batch is in the same order.
+func TestLedgerForEvaluatorLookupMultipleAccounts(t *testing.T) {
+	db, shutdownFunc := setupPostgres(t)
+	defer shutdownFunc()
+
+	addAccountQuery :=
+		"INSERT INTO account " +
+			"(addr, microalgos, rewardsbase, rewards_total, deleted, created_at) " +
+			"VALUES ($1, 0, 0, 0, false, 0)"
+	addAccountAssetQuery :=
+		"INSERT INTO account_asset (addr, assetid, amount, frozen, deleted, created_at) " +
+			"VALUES ($1, $2, 0, false, false, 0)"
+	addAssetQuery :=
+		"INSERT INTO asset (index, creator_addr, params, deleted, created_at) " +
+			"VALUES ($1, $2, '{}', false, 0)"
+	addAppQuery :=
+		"INSERT INTO app (index, creator, params, deleted, created_at) " +
+			"VALUES ($1, $2, '{}', false, 0)"
+	addAccountAppQuery :=
+		"INSERT INTO account_app (addr, app, localstate, deleted, created_at) " +
+			"VALUES ($1, $2, '{}', false, 0)"
+
+	addresses := []basics.Address{
+		test.AccountA, test.AccountB, test.AccountC, test.AccountD, test.AccountE}
+	seq := []int{4, 9, 3, 6, 5, 1}
+
+	for i, address := range addresses {
+		_, err := db.Exec(context.Background(), addAccountQuery, address[:])
+		require.NoError(t, err)
+
+		// Insert all types of creatables. Note that no creatable id is ever repeated.
+		for j := range seq {
+			_, err = db.Exec(
+				context.Background(), addAccountAssetQuery, address[:], i + 10*j + 100)
+			require.NoError(t, err)
+
+			_, err = db.Exec(
+				context.Background(), addAssetQuery, i + 10*j + 200, address[:])
+			require.NoError(t, err)
+
+			_, err = db.Exec(
+				context.Background(), addAppQuery, i + 10*j + 300, address[:])
+			require.NoError(t, err)
+
+			_, err = db.Exec(
+				context.Background(), addAccountAppQuery, address[:], i + 10*j + 400)
+			require.NoError(t, err)
+		}
+	}
+
+	tx, err := db.BeginTx(context.Background(), readonlyRepeatableRead)
+	require.NoError(t, err)
+	defer tx.Rollback(context.Background())
+
+	l, err := ledger_for_evaluator.MakeLedgerForEvaluator(
+		tx, crypto.Digest{}, transactions.SpecialAddresses{})
+	require.NoError(t, err)
+	defer l.Close()
+
+	{
+		addressesMap := make(map[basics.Address]struct{})
+		for _, address := range addresses {
+			addressesMap[address] = struct{}{}
+		}
+		err := l.PreloadAccounts(addressesMap)
```

Comment:
Could you leave a comment that `PreloadAccounts` is the function being tested and specifically we're verifying that `batch.Queue` and `batch.Result` are in the correct order?

Reply:
done